Added test_nofs_misc.cpp for tsk/fs/nofs_misc.cpp, Updated Makefile.am

singhpratik5 opened the pull request:
Added test_nofs_misc.cpp for tsk/fs/nofs_misc.cpp
Updated Makefile.am accordingly
Removed ils test content for now

bcarrier committed:
Do not consider it an error if there is no idx_root

bcarrier committed:
fix null pointer issues

gdicristofaro committed:
fix for non-windows builds

(cherry picked from commit 36e92fb48cdebbfd5a64a0a559e964782c067d6e)

gdicristofaro committed:
revert unnecessary change

(cherry picked from commit 0b5c6b5363a46f887d8e8d29872ca91633fb3021)

gdicristofaro committed:
fix for PRIcTSK on non-windows machines

(cherry picked from commit 3de274cdb948cbc3b279930d66c5fe7f66f27db7)

gdicristofaro committed:
updates for password in java code

(cherry picked from commit 16dcd47441a3fa2f439c4e8fc204ddc676718776)

gdicristofaro committed:
updates for persistence

(cherry picked from commit ab2a256385cf5c53947aa180046e6c28027d7ad4)

gdicristofaro committed:
updates for bridge

(cherry picked from commit 16fa4967d594ff8925366bdd4ef03e57aa7c45c6)

APriestman committed:
Testing

(cherry picked from commit bf0e4fb38b8643e8b162fc84b0ec010bb07aaa5e)

gdicristofaro committed:
updates

(cherry picked from commit 4090d4bd5f110896474cf2f74c009e663a511222)

gdicristofaro committed:
switch to _decrypt

(cherry picked from commit f9faedff312cc82fefe587c3f1eb8fe36de7b6cb)

gdicristofaro committed:
minor fix

(cherry picked from commit 3fb4c6b4186f82eba984d7234dbab356bded7588)

APriestman committed:
Testing caching improvements

(cherry picked from commit 8431d6f8c22c487ba2cde7a147c4a61b4abbe55e)

gdicristofaro committed:
update for single threaded bitlocker decrypting

(cherry picked from commit 4abe7994a1bbcbfdd451dc119da847486218b2ae)

gdicristofaro committed:
method name update

(cherry picked from commit b03fdc0b456d169c9c6df26140147d0dd94a55a8)

APriestman committed:
Update logical dir caching

(cherry picked from commit f29bfb48088cb7362c37fd3c0bfb0411e9b92a84)

APriestman committed:
Cleanup

(cherry picked from commit 6ebeb6f2bb3d21440ad9ec7c7b1626f3bbc03591)

APriestman committed:
Increase max size of logical file sets

(cherry picked from commit aa68d6f308115e27a6bdfa6c060f4aabb413e1bb)

APriestman committed:
Cleanup

(cherry picked from commit 11de4d2080590986a3c82e0daa286b45dc07f096)

gdicristofaro committed:
change to const method

(cherry picked from commit b367397ca6ead50baf0929c9227c96e33d5a4c18)

APriestman committed:
Fix parens

(cherry picked from commit b76919d12d2ad6b39178dbd35cf6ff98c3f6905e)

APriestman committed:
Cache selected folders when opening the file system.
Don't put duplicates in the cache.

(cherry picked from commit cd8f3e3dc7c14f89fbcf443820e74d1376248df9)

eugene7646 committed:
TSK changes

(cherry picked from commit b9005e05b9a0bb9b15f0bd9bdb1c27a3dbbac300)

gdicristofaro committed:
updates to avoid sql injection

(cherry picked from commit 41e3c5474eaaca239010469e77f5a597d258adde)

jayaramcs committed:
Setting Journal mode to WAL in sqlite

(cherry picked from commit fc71f3e65e07d75f3000968ebe3a159295cb0ffc)

jayaramcs committed:
Upgrading libs to latest versions

(cherry picked from commit 4236633ecb9dd1602170ef4faf0010771fa1e0d2)

jayaramcs committed:
New open/load case apis that take a boolean useWAL option.

(cherry picked from commit 4ee082761494d607bb57bcb9f51e873283a2b7b9)

jayaramcs committed:
Comment updates

(cherry picked from commit 05db986ea4bec66fea4f1750f6d103914653c10f)

Commit:
renamed clean_java to clean-java for consistency. added check-java

Commit:
check-java added as a Makefile target. it works on ubuntu.

gdicristofaro committed:
sqlite optimizations testing

bcarrier committed:
New version files for 4.14.0

bcarrier committed:
Updated NEWS

gdicristofaro committed:
updates for sqlite optimize to run max every 5 minutes

jayaramcs committed:
Merge pull request #9 from APriestman/cherryPickTest4

Merge SKL commits since Dec 3

gdicristofaro committed:
updates to call optimize for each connection

gdicristofaro committed:
Merge branch 'release-4.14.0' of github.com:SleuthKitLabs/sleuthkit into CT-10027d_callOptimizeForEachConn

jayaramcs committed:
Merge pull request #10 from gdicristofaro/CT-10027d_callOptimizeForEachConn

CT-10027 (d) call optimize for each conn

bcarrier committed:
Merge pull request #3257 from sleuthkit/release-4.14.0-patch-0404

Changes from SKL to CT-tag

simsong committed:
added ci/cd from develop to release-4.14.0

simsong committed:
added branch

simsong committed:
removed -Werror

simsong committed:
add test_loaddb.sh, which is the test we care about

simsong committed:
just run the tsk_loaddb test

simsong committed:
re-enabled CODECOV upload; removed mingw

simsong committed:
forgot to call make clean and make again

gdicristofaro committed:
updates for case db access manager optimize statement

simsong committed:
upgraded appveyor

simsong committed:
improving tests

jayaramcs committed:
Merge pull request #13 from gdicristofaro/CT-10027h_caseDbAccessManager

CT-10027 (h) updates for case db access manager optimize statement

gdicristofaro committed:
optimize commenting

bcarrier committed:
Merge pull request #14 from gdicristofaro/CT-10027i_optimizeComment

CT-10027 (i) optimize comment

bcarrier committed:
Merge remote-tracking branch 'sleuthkitlabs/release-4.14.0' into release-4.14.0-patch-0410

bcarrier committed:
Merge pull request #3265 from SleuthKitLabs/release-4.14.0-patch-0410

Move the new call to optimize

bcarrier committed:
Merge pull request #3259 from sleuthkit/release-4.14.0-add-cicd

add ci/cd from develop to release-4.14.0

bcarrier committed:
updated JDK docs and debian dependiencies

bcarrier committed:
update sql for debian

bcarrier committed:
Merge branch 'release-4.14.0' of github.com:sleuthkit/sleuthkit into release-4.14.0

bcarrier committed:
Added make check

bcarrier committed:
.

bcarrier committed:
New version files for 4.14.0

bcarrier committed:
added ident

bcarrier committed:
Merge branch 'release-4.14.0' of github.com:sleuthkit/sleuthkit into release-4.14.0

bcarrier committed:
added VS2019

bcarrier committed:
.

bcarrier committed:
.

bcarrier committed:
.

bcarrier committed:
.

bcarrier committed:
.

APriestman committed:
Change .cend() to .end()

AburtoArielPM committed:
Agregar Bundle_es.properties con traducción al español

APriestman committed:
Record any large dirs that are skipped

jayaramcs committed:
Merge pull request #15 from APriestman/removeCend

Change .cend() to .end()

jayaramcs committed:
Merge pull request #3283 from SleuthKitLabs/develop-4

[Bug fix] Change .cend() to .end()

APriestman committed:
Add tsk error listener

APriestman committed:
Cleanup

jayaramcs committed:
Merge pull request #16 from APriestman/recordLargeDirs

Add optional error callback

APriestman committed:
Record a large dir error if nrd.allocsize is very large

jayaramcs committed:
Merge pull request #17 from APriestman/10206_recordNtfsLargeFileError

Record a large dir error if nrd.allocsize is very large

Taha-Ebrahim committed:
test crypto again

jayaramcs committed:
Merge pull request #18 from SleuthKitLabs/release-4.14.2

Release 4.14.2

jayaramcs committed:
Merge pull request #3300 from SleuthKitLabs/develop-4

Stability improvements / Error propagation when traversing 1M+ files in directory

Taha-Ebrahim committed:
Merge pull request #12 from Taha-Ebrahim/new-crypto

Taha-Ebrahim committed:
Update build-unix.yml

Taha-Ebrahim committed:
Merge pull request #14 from Taha-Ebrahim/Taha-Ebrahim-patch-1-1

Update build-unix.yml

Taha-Ebrahim committed:
Update code-coverage.yml

Taha-Ebrahim committed:
added fatfs tests

Taha-Ebrahim committed:
redo fatfs tests

Taha-Ebrahim committed:
changes to test_fatfs

Taha-Ebrahim committed:
memory management

Taha-Ebrahim committed:
check whether test_fatfs running

Taha-Ebrahim committed:
fix tests

Taha-Ebrahim committed:
Merge pull request #15 from Taha-Ebrahim/test-fs

added test-fs tests

Taha-Ebrahim committed:
Update build-unix.yml

Taha-Ebrahim committed:
Update code-coverage.yml

Taha-Ebrahim committed:
fixed data

Taha-Ebrahim committed:
Merge pull request #16 from Taha-Ebrahim/fix-data

fixed data

singhpratik5 committed:
Added test_nofs_misc.cpp for tsk/fs/nofs_misc.cpp, Updated Makefile.am accordingly Removed ils test content for now

singhpratik5 committed:
Added test_nofs_misc.cpp for tsk/fs/nofs_misc.cpp, Updated Makefile.am accordingly Removed ils test content for now

singhpratik5 committed:
Merge branch 'develop' into experiment

Taha-Ebrahim committed:
changes for lvm and increasing code resiliency

Taha-Ebrahim committed:
get environment fix

Taha-Ebrahim committed:
add another null check

Taha-Ebrahim committed:
added graceful skip

Commit:
Fix typos. Use get() and don't call close manually for std::unique_ptrs.

singhpratik5 committed:
Update test_nofs_misc.cpp 

Fixed a test case to use tmpfile for Mingw.

singhpratik5 committed:
Update test_nofs_misc.cpp 

Fixed a test case to use tmpfile for Mingw.

simsong commented:
@singhpratik5  — it looks like `strstr` under mingw & wine is returning "" for string not found rather than `nullptr`. See:

<img width="846" alt="image" src="https://github.com/user-attachments/assets/6a355309-0e72-4bc9-96c6-301fe2784ef5" />

I've looked online and haven't been able to verify this one way or another. Can you try putting some print statements in and running the test manually? In the meantime, so that you can keep busy, I recommend moving to another test case.  Perhaps `blkls` or `blkstat`. See https://app.codecov.io/github/singhpratik5/sleuthkit/tree/develop/tools%2Ffstools

Sorry that this has been so much work and is so frustrating, but I am sure that this will all come together in a week or two.

simsong committed:
Merge pull request #3308 from uckelman-sf/fix_libvslvm_typos

Fix typos in code built with libvslvm

simsong committed:
Merge branch 'develop' into apfs-fix

simsong committed:
Merge pull request #3280 from AburtoArielPM/traduccion-es

Agregar Bundle_es.properties con traducción al español

Taha-Ebrahim committed:
updated encryption docs

Taha-Ebrahim committed:
updated encryption docs

Taha-Ebrahim committed:
Merge branch 'develop' into apfs-fix

singhpratik5 commented:
@simsong Fine, I will put some print statements and check it locally. 
I will start working on 'blkls' too.

singhpratik5 commented:
@simsong 
![image](https://github.com/user-attachments/assets/f5f528af-e0b3-4ced-9c30-66739a05431b)
Here is the output for the failing one. 
I have added two comments to see actual data. Should I push it to check for Mingw, I tried to run manually using wine on local, but it didn't work. 

test/tsk/fs/test_nofs_misc.cpp:44: PASSED:
  REQUIRE( strstr(buffer, "Data") != nullptr )
with expansion:
  "Data
  Block Size: 4096
  Block Range: 0 - 1234
  "
  !=
  nullptr

Commit:
img must not be used after move.

singhpratik5 committed:
Added print stmt to test_nofs_misc.cpp

singhpratik5 committed:
Added print stmt to test_nofs_misc.cpp

simsong committed:
Merge pull request #3309 from uckelman-sf/fix_use_afer_move

Do not use img after move

simsong commented:
I looked at the output of the tests. I could find this:
```
..............................................................................

test/tsk/fs/test_nofs_misc.cpp:55: FAILED:
  REQUIRE( strstr(buffer, "Data") != nullptr )
with expansion:
  {null string} != nullptr

===============================================================================
test cases:  81 |  80 passed | 1 failed
assertions: 333 | 332 passed | 1 failed
```
but I could not easily find where `buffer` is printed or the results of the `strstr` are printed. (Print both with %s and with %x — that is, both show the string and pointer value)

singhpratik5 commented:
@simsong I didn't push the update on remote. I am pushing it now.

APriestman committed:
Look for logical dir sibling in cache to speed up loading

singhpratik5 committed:
Updated usage: Removed full path printing.
Added just command name to get printed instead.

singhpratik5 committed:
Updated usage: Removed full path printing.

Added just command name to get printed instead.

simsong committed:
Merge pull request #3310 from singhpratik5/fix-usage-messages

Fixed usage messages to print command name only

APriestman committed:
Release lock on error

singhpratik5 committed:
Merge remote-tracking branch 'upstream/develop' into develop

Taha-Ebrahim committed:
reverted changes ref #3309

Taha-Ebrahim committed:
remove testdb journal

singhpratik5 committed:
Added test_fs_types.cpp and test_fs_name.cpp Updated makefile.am accordingly

simsong committed:
Merge branch 'develop' into apfs-fix

singhpratik5 committed:
Merge branch 'develop' into fs-tests

simsong committed:
Update copyright

simsong committed:
Merge pull request #3313 from sleuthkit/simsong-patch-1

Update copyright

DavidKorczynski committed:
ossfuzz: fix build

Signed-off-by: David Korczynski <david@adalogics.com>

Taha-Ebrahim committed:
minor cleanup

Taha-Ebrahim committed:
minor cleanup

simsong committed:
Merge branch 'develop' into apfs-fix

simsong committed:
Merge pull request #3314 from DavidKorczynski/fix-ossfuzz

ossfuzz: fix build

simsong committed:
Merge pull request #3307 from Taha-Ebrahim/apfs-fix

add tests for fatfs.cpp, crypto.cpp, fix LVM support

singhpratik5 committed:
Update test_fs_name.cpp

Commented out cases with invalid tag.

singhpratik5 committed:
Merge remote-tracking branch 'upstream/develop' into fs-tests

singhpratik5 committed:
Removed test_fls.cpp
	modified:   Makefile.am
	deleted:    test/tsk/fs/test_fls.cpp

Taha-Ebrahim committed:
amended testing plan

Taha-Ebrahim committed:
minor changes

simsong committed:
Merge remote-tracking branch 'origin/develop' into pratik5-develop.
Bring to sleuthkit/develop.

singhpratik5 committed:
Merge pull request #3 from singhpratik5/pratik5-develop

Pratik5 develop

singhpratik5 committed:
Merge branch 'develop' into fs-tests

singhpratik5 committed:
Update test_fs_types.cpp

used _TSK_T macro instead of strings as char.
Updated a str name as per it's actual behaviour

simsong committed:
Merge pull request #2 from singhpratik5/fs-tests

Added test_fs_types.cpp and test_fs_name.cpp Updated makefile.am acco…

simsong committed:
comment out reference to internal symbol

simsong committed:
Update AppVeyor config to run on all PRs without duplicate jobs

simsong committed:
Merge pull request #3319 from sleuthkit/fix-tzname

Comment out reference to internal `__tzinfo` symbol and fix AppVeyor redundancy

simsong committed:
Merge branch 'develop' into develop

Taha-Ebrahim committed:
appveyor test data fix

jayaramcs committed:
Merge pull request #19 from APriestman/10357_logicalFileImprovement

Look for logical dir sibling in cache to speed up loading

simsong committed:
make github actions run on PR to develop-4.14

simsong committed:
Merge pull request #3322 from sleuthkit/expand-actions

make github actions run on PR to develop-4.14

simsong committed:
minor change to force actions to run

simsong committed:
Merge branch 'develop-4.14' into pr-3321

simsong committed:
whoops; specified wrong branch

simsong committed:
Merge remote-tracking branch 'origin/develop-4.14' into pr-3321

simsong committed:
Merge remote-tracking branch 'origin/pr-3321' into pr-3321

simsong committed:
Trigger CI

simsong committed:
Merge pull request #3320 from Taha-Ebrahim/appveyor-data-fix

AppVeyor Test Data Directory Fix

simsong committed:
Merge branch 'develop' into develop

APriestman committed:
Fix non-windows compile issues

APriestman committed:
Change to XP-compatible toLower

jayaramcs committed:
Merge pull request #20 from APriestman/10357_fixNonWindowsBuild

Fix non-windows compile issues

simsong committed:
Merge pull request #3321 from SleuthKitLabs/develop-4.14

Logical File Improvements

simsong committed:
Merge pull request #3316 from Taha-Ebrahim/cli-runner

Updated CLI test runner proposal

simsong committed:
make github action run on develop-4.14 branch

simsong committed:
Update build-unix.yml

make github action run on develop-4.14 branch

simsong committed:
merged in changes from @brianc to new develop

simsong committed:
added header for std::towlower

simsong committed:
sprintf -> snprintf

simsong committed:
sprintf -> snprintf

simsong committed:
remove unneeded variable

simsong committed:
remove assignment to variable that is then ignored

simsong committed:
sprintf -> snprintf

simsong committed:
sprintf -> snprintf

simsong committed:
sprintf -> snprintf

simsong committed:
clean buffer code and sprintf -> snprintf

simsong committed:
sprintf -> snprintf

simsong committed:
sprintf -> snprintf

simsong committed:
sprintf -> snprintf

simsong committed:
sprintf -> snprintf

simsong committed:
fix case_insensitive_compare for non-Windows

simsong committed:
Merge pull request #3325 from sleuthkit/simsong-patch-1

Fix develop-4.14 to compile under windows & mac; Removes many warnings;

simsong committed:
fixed CI/CD for develop-4.14

simsong committed:
Merge remote-tracking branch 'origin/develop-4.14' into develop-4.14

simsong committed:
merged develop-4.14 into develop. May not work

singhpratik5 committed:
Added tests for tsk/fs/unix_misc.cpp, Added test_unix_misc.cpp to Makefile.am

singhpratik5 committed:
minor _TSK_T macro fix

singhpratik5 committed:
Minor changes in test_unix_misc.cpp

simsong committed:
Merge pull request #3330 from singhpratik5/fs-test

Added tests for tsk/fs/unix_misc.cpp

simsong committed:
merged all changes from develop-4.14 and resolved all conficts. All checks pass

simsong committed:
removed <<<<< HEAD stuff

simsong committed:
change 4.15.0-dev to 4.15.0-develop for consistency

simsong committed:
added [[maybe_unused]] to curImgInfo parameter

simsong committed:
added [[maybe_unused]] to wrong place

simsong committed:
enable undefined sanitizer in ci/cd

simsong committed:
Merge branch 'develop' into develop-take-develop-4.14

simsong committed:
Merge pull request #3332 from sleuthkit/develop-take-develop-4.14

Take develop 4.14 recent commits to develop

simsong committed:
Merge branch 'develop' into ignore_idx_root

simsong committed:
Merge branch 'develop-4.14' into pr-3321

simsong committed:
Merge branch 'develop' into pr-3321

simsong committed:
Merge pull request #3326 from sleuthkit/ignore_idx_root

Ignore idx root

simsong committed:
Merge pull request #3323 from sleuthkit/pr-3321

Pr 3321 to develop

simsong committed:
updated

simsong committed:
just add the coverage_local.sh test

simsong committed:
Merge branch 'develop' into release-4-13.0-patch

simsong committed:
Update Makefile.am

restored accidentally deleted endif

simsong committed:
Merge pull request #3333 from sleuthkit/patch-improve-test-coverage

Patch - add a script to run the test coverage locally

simsong committed:
Merge pull request #3334 from sleuthkit/release-4-13.0-patch

add check-java target to Makefile.am

singhpratik5 committed:
Merge branch 'develop' into develop

simsong committed:
Merge pull request #3318 from singhpratik5/develop

Test files for tsk/fs directory

singhpratik5 committed:
Merge remote-tracking branch 'refs/remotes/origin/experiment' into experiment

singhpratik5 committed:
Update test_nofs_misc.cpp

Remove one test

singhpratik5 commented:
@simsong should I merge it in my develop or directly open a PR from experiment to sleuthkit/develop?

simsong commented:
> @simsong should I merge it in my develop or directly open a PR from experiment to sleuthkit/develop?

Probably not?

I recommend bringing your develop up to date from sleuthkit/develop, and then seeing what happens to this one. In general, I would keep your experiments separate from your develop, and keep your develop in sync with sleuthkit/develop.

singhpratik5 commented:
My develop is already up to date with sleuthkit/develop. I think it would be better to open PR from experiment to sleuthkit/develop for now.
Here is link: https://github.com/sleuthkit/sleuthkit/compare/develop...singhpratik5:sleuthkit:experiment
![image](https://github.com/user-attachments/assets/e34c3804-943b-483d-9a64-4e360e8835b5)

simsong commented:
I don't think that your develop is up-to-date with SleuthKit's develop. This PR has 81 files being changed and commits from `bcarrier` and many others. If your develop was up-to-date with sleuthkit's develop, then they wouldn't be there.

I'm not sure what is going on, but you should definately put this on hold and separately do a PR from sleuthkit's develop to this one.

<img width="856" alt="image" src="https://github.com/user-attachments/assets/a9f2976d-b178-4bb9-9abf-ef32b46822b3" />

simsong commented:
I suspect that this develop is up-to-date with Sleuthkit's `main` and that your `experiment` is up-to-date with sleuthkit's `develop`.

simsong commented:
Okay,  you may be right. Your develop does in fact appear to be up-to-date with sleuthkti develop. So just merge this in and see what happens!

singhpratik5 commented:
The reason for this mess is rebase from sleuthkit/develop to experiment that I did to sync experiment with sleuthkit/develop. 
I think it would be better to open a PR to sleuthkit/develop as only 3 files are getting changed in that and then I would delete branch experiment. See last screenshot attached.
![image](https://github.com/user-attachments/assets/cfcb0ae2-f4c3-42a5-9583-34e08ec216b1)
![Screenshot 2025-07-07 210654](https://github.com/user-attachments/assets/80c25f52-ac8f-4eac-900b-7092ddf7186e)
![image](https://github.com/user-attachments/assets/3e095e60-5412-4160-b22f-0b655d7cfb82)

simsong commented:
Indeed, rebase is a pretty powerful tool... I recommend avoiding it for now.

On Mon, Jul 7, 2025 at 11:43 AM singhpratik5 ***@***.***>
wrote:

> *singhpratik5* left a comment (singhpratik5/sleuthkit#1)
> <https://github.com/singhpratik5/sleuthkit/pull/1#issuecomment-3045669897>
>
> The reason for this mess is rebase from sleuthkit/develop to experiment
> that I did to sync experiment with sleuthkit/develop.
> I think it would be better to open a PR to sleuthkit/develop as only 3
> files are getting changed in that and then I would delete branch
> experiment. See last screenshot attached.
> image.png (view on web)
> <https://github.com/user-attachments/assets/cfcb0ae2-f4c3-42a5-9583-34e08ec216b1>
> Screenshot.2025-07-07.210654.png (view on web)
> <https://github.com/user-attachments/assets/80c25f52-ac8f-4eac-900b-7092ddf7186e>
> image.png (view on web)
> <https://github.com/user-attachments/assets/3e095e60-5412-4160-b22f-0b655d7cfb82>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/singhpratik5/sleuthkit/pull/1#issuecomment-3045669897>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMFHLAU4N5ES7V4MC6KSM33HKIRXAVCNFSM6AAAAAB7NH4MHSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTANBVGY3DSOBZG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

singhpratik5 committed:
Update test_nofs_misc.cpp

Added file description and sign
